app/src/main/java/com/fivegmag/a5gmsmediasessionhandler/MediaSessionHandlerMessengerService.kt

```diff
@@ -302,5 +328,128 @@ class MediaSessionHandlerMessengerService() : Service() {
         return mMessenger.binder
     }
 
+    /* Refer to TS26.512 Clause 4.7.4
+    The Service Access Information indicating whether Consumption Reporting is provisioned for downlink streaming
+    sessions is described in clause 11.2.3. When the clientConsumptionReportingConfiguration.samplePercentage
+    value is 100, the Media Session Handler shall activate the consumption reporting procedure.
+    If the samplePercentage is less than 100, the Media Session Handler shall generate a random number
+    which is uniformly distributed in the range of 0 to 100, and the Media Session Handler shall activate
+    the consumption report procedure when the generated random number is of a lower value than
+    the samplePercentage value.The Service Access Information indicating whether Consumption Reporting
+    is provisioned for downlink streaming sessions is described in clause 11.2.3. When the
+    clientConsumptionReportingConfiguration.samplePercentage value is 100, the Media Session Handler
+    shall activate the consumption reporting procedure. If the samplePercentage is less than 100,
+    the Media Session Handler shall generate a random number which is uniformly distributed in
+    the range of 0 to 100, and the Media Session Handler shall activate the consumption report
+    procedure when the generated random number is of a lower value than the samplePercentage value.
+    The Service Access Information indicating whether Consumption Reporting is provisioned for downlink streaming sessions is described in clause 11.2.3. When the clientConsumptionReportingConfiguration.samplePercentage value is 100, the Media Session Handler shall activate the consumption reporting procedure. If the samplePercentage is less than 100, the Media Session Handler shall generate a random number which is uniformly distributed in the range of 0 to 100, and the Media Session Handler shall activate the consumption report procedure when the generated random number is of a lower value than the samplePercentage value.
+     */
+    private fun IsConsumptionReportingActivated(): Boolean {
+        //if(currentServiceAccessInformation.isInitialized) {
+        if(currentServiceAccessInformation == null) {
+            return false
+        }
+
+        var samplePercentage: UInt =  currentServiceAccessInformation!!.clientConsumptionReportingConfiguration.samplePercentage;
+        if(samplePercentage > SamplePercentage_100.toUInt())
+        {
+            Log.i(TAG, "shilin>>>Invaild samplePercentage $samplePercentage in currentServiceAccessInformation")
+            return false;
+        }
+
+        if(SamplePercentage_100.toUInt() == samplePercentage)
```

Comment:
Use this function to check if reporting should be activated or not according to `samplePercentage`:

````
      private fun shouldReportAccordingToSamplePercentage(samplePercentage: Float?): Boolean {
            if (samplePercentage != null && samplePercentage <= 0) {
                return false
            }

            if (samplePercentage == null || samplePercentage >= 100.0) {
                return true
            }

            return utils.generateRandomFloat() < samplePercentage

        }
````